File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        if (len(nums)==1):
            return 1
        L=0
        for R in range(1,len(nums)):
            if (nums[R]!=nums[R-1]):
                L+=1
                nums[L]=nums[R]
        return L+1

File: Data Structures & Algorithms/string-encode-and-decode/submission-0.go
type Solution struct{}

func (s *Solution) Encode(strs []string) string {
	var builder strings.Builder
	for _, val := range strs {
		builder.WriteString(strconv.Itoa(len(val)))
		builder.WriteString("#")
		builder.WriteString(val)
	}
	return builder.String()
}


func (s *Solution) Decode(encoded string) []string {
	var res []string
	for i:=0;i<len(encoded); {
		j := strings.Index(encoded[i:],"#")
		l, _ := strconv.Atoi(encoded[i:i+j])
		res = append(res,encoded[i+j+1:i+j+1+l])
		i += j+l+1
	}
	return res

}
